src/MOI_wrapper.jl

```diff
@@ -1467,6 +1467,29 @@ function MOI.add_constraints(
     return indices
 end
 
+function MOI.delete(
+    model::Optimizer,
+    cs::Vector{<:MOI.ConstraintIndex{MOI.ScalarAffineFunction{Float64}, <:Any}}
+)
+    # The `sort!`s called are necessary for improving the efficiency of the
+    # updates (i.e., without them we cannot use `searchsorted*` methods).
+    _update_if_necessary(model)
+    rows_to_delete = sort!(getfield.(_info.(model, cs), :row))
+    del_constrs!(model.inner, rows_to_delete)
+    _require_update(model)
+    for (_, info) in model.affine_constraint_info
+        info.row -= searchsortedlast(rows_to_delete, info.row)
```

Comment:
I would add a comment explaining what this loop does. IIUC, for alll constraints, it gets the index of the last row with row index smaller. That gives the number of rows before, hence substracting this give the new row index

Reply:
I will do it. I thought about writing this comment already. However I have already used this trick in other places (I think in both Gurobi.jl and CPLEX.jl) and did not comment it. Should I add it on all these places?

Reply:
Yes, it would help there also IMO

Reply:
Created a PR for [Gurobi.jl](https://github.com/JuliaOpt/Gurobi.jl/pull/314) and another for [CPLEX.jl](https://github.com/JuliaOpt/CPLEX.jl/pull/298).